packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1438,9 +1441,36 @@ class _NestedScrollPosition extends ScrollPosition implements ScrollActivityDele
     coordinator.updateCanDrag();
   }
 
-  void updateCanDrag(double totalExtent) {
-    context.setCanDrag(physics.allowUserScrolling &&
-        (totalExtent > (viewportDimension - maxScrollExtent) || minScrollExtent != maxScrollExtent));
+  void updateCanDrag({
+    required double totalInnerExtent,
+    required double overlapExtent
+  }) {
+    // This is only called for the outer position.
+    // Any overlap extent is already deducted from the SliverGeometry of any
+    // SliverOverlapAbsorber wrapped slivers, so the maxExtent of the outer
+    // position does not reflect the extent that has been absorbed. This is as
+    // expected since the inner and outer scroll views' extents are manipulated
+    // in this way for them to behave as one, but the extent must be accounted
+    // for when considering the full extent available to drag.
+    // For example, a pinned SliverAppBar in the outer position with a minExtent
+    // and maxExtent of 56.0 will have a ScrollPosition.maxExtent of 0.0 after
+    // the SliverOverlap is applied. This is correct because the pinned sliver
+    // app bar should not have to consume scrolling before the inner body of the
+    // NestedScrollView scrolls. However, when calculating the drag here, 0
+    // could result in scrolling being disabled unexpectedly if the inner scroll
+    // view by itself were not long enough to allow scrolling on its own.
+    final double maxScrollExtent = this.maxScrollExtent + overlapExtent;
+    final bool extentAllows =
+      // There is already room to scroll in the outer scroll view.
+      minScrollExtent != maxScrollExtent
+      // The inner extent exceeds the bounds of the viewport, accounting for the
+      // outer extent.
+      || totalInnerExtent > (viewportDimension - maxScrollExtent)
```

Comment:
If [this]( https://github.com/flutter/flutter/pull/127718/files#r1209928710) is right, we just should change `extentAllows ` to 
```dart
final bool extentAllows = this.minScrollExtent != this.maxScrollExtent || totalInnerExtent > 0
```

Reply:
So I realized this is not accounting for the physics at all! Which reminded me of https://github.com/flutter/flutter/issues/82391... so I changed it to what i think it should really be. Even if the extent is not enough to scroll, I don't want to completely ignore AlwaysScrollableScrollPhysics, which should always scroll even if it is not long enough.